charmhub/info.go

```diff
@@ -38,5 +39,16 @@ func (c *InfoClient) Info(ctx context.Context, name string) (transport.InfoRespo
 	if err := c.client.Get(ctx, path, &resp); err != nil {
 		return resp, errors.Trace(err)
 	}
+
+	if len(resp.ErrorList) > 0 {
+		var combined []string
+		for _, err := range resp.ErrorList {
+			if err.Message != "" {
+				combined = append(combined, err.Message)
+			}
+		}
+		return resp, errors.Errorf(strings.Join(combined, "\n"))
+	}
```

Comment:
This could be a helper function, I believe that each api call will end up with []APIError to deal with.